### DISCUSSION TITLE
How can I change the color of a Color Switch?

### DISCUSSION THREAD

**Discussion by normanth** [zwave-js/node-red-contrib-zwave-js#290] [Q&A] [answered]
Hi,

I hope someone has a tip for me. I just can't get any further with this.

I only want to set a color to my Color Switch (f.e. hexColor).
But I don't know, what I have to set in the ZWave CMD Factory.

API: CCAPI
CC: Color Switch
Method: set (or updateCurrentColor?)
Endpoint: 0
Params [ ]: ?

In the Parameter Definition I find:

```
async set(
    options: ColorSwitchCCSetOptions,
): Promise<SupervisionResult | undefined>;
```

But what is "ColorSwitchCCSetOptions" and how I can set this in the ZWave CMD Factory?

Thanks a lot!
Norman

**Comment by normanth** [comment]
I found this:

```
export type ColorSwitchCCSetOptions = (ColorTable | { hexColor: string }) & {
	duration?: Duration | string;
};
```

?!?!

**Comment by marcus-j-davies** [comment, marked as answer]
Hi @normanth 

For that, you will use something like the bellow

```
CC: Color Switch
Method: Set
Params: [{'hexColor':'#374b96'}]
```

The `Params` input supports `JSOnata`. -so you can pass in a msg property for the colour also
```
Params: [{'hexColor':payload.color}]
```

The params are an array, of any object (based on the CC/Method being executed)

But, to be honest you should really be using the ValueAPI, as its superior to the CC API

**Reply by marcus-j-davies** [reply]
And here is the Options object.

https://github.com/zwave-js/node-zwave-js/blob/c21f13734d3283fde6dc5ede1535bc6be33c12f3/packages/cc/src/cc/ColorSwitchCC.ts#L946

**Reply by normanth** [reply]
Thank you very much!!
